### PR TITLE
Refactor delaying of staking updates

### PR DESCRIPTION
**Pull request by jinmannwong**
Refactor delaying of staking updates into own file to minimise changes to other sdk modules

### CODE REVIEW

File: x/staking/keeper/delay_updates.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package keeper
+
```
Comment (jinmannwong) [inline x/staking/keeper/delay_updates.go:2]: Main changes are the refactoring of validator update delay checks into this file.

File: x/staking/handler_test.go [hunk 2]
```diff
@@ -95,9 +95,9 @@ func TestValidatorByPowerIndex(t *testing.T) {
 
 	ctx = ctx.WithBlockTime(finishTime)
 	header := abci.Header{Entropy: testBlockEntropy()}
-	BeginBlocker(ctx, abci.RequestBeginBlock{Header: header}, &keeper)
-	EndBlocker(ctx, &keeper)
-	EndBlocker(ctx, &keeper)
+	BeginBlocker(ctx, abci.RequestBeginBlock{Header: header}, keeper)
+	EndBlocker(ctx, keeper)
+	EndBlocker(ctx, keeper)
```
Comment (ejfitzgerald) [inline x/staking/handler_test.go:100]: Double end blocker - intentional?

File: x/staking/keeper/delay_updates.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var (
```
Comment (ejfitzgerald) [inline x/staking/keeper/delay_updates.go:9]: does it makes sense for this to be const

Comment (jinmannwong) [reply x/staking/keeper/delay_updates.go]: Yes, that would be better.

File: x/staking/handler_test.go (hunk 2 above)
Comment (jinmannwong) [reply x/staking/handler_test.go]: It was in the original test, but the test seems to run fine without it so it was probably just a typo.

**Pull request merged**